sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/models/KeyVaultRoleDefinition.java

```diff
@@ -13,23 +15,36 @@
     private final String id;
     private final String name;
     private final KeyVaultRoleDefinitionType type;
-    private final KeyVaultRoleDefinitionProperties properties;
+    private final String roleName;
+    private final String description;
+    private final KeyVaultRoleType roleType;
+    private final List<KeyVaultPermission> permissions;
+    private final List<KeyVaultRoleScope> assignableScopes;
 
     /**
      * Creates a new {@link KeyVaultRoleDefinition role definition} with the specified details.
      *
      * @param id The ID for this {@link KeyVaultRoleDefinition role definition}.
      * @param name The name for this {@link KeyVaultRoleDefinition role definition}.
      * @param type The type of this {@link KeyVaultRoleDefinition role definition}.
-     * @param properties {@link KeyVaultRoleDefinitionProperties properties} of this {@link KeyVaultRoleDefinition
-     * role definition}.
+     * @param roleName The name of the role.
+     * @param roleDescription The description of the role.
+     * @param roleType The type of the role.
+     * @param permissions The {@link KeyVaultPermission permissions} the {@link KeyVaultRoleDefinition role definition}
+     * has.
+     * @param assignableScopes The assignable scopes of the {@link KeyVaultRoleDefinition role definition}.
      */
-    public KeyVaultRoleDefinition(String id, String name, KeyVaultRoleDefinitionType type,
-                                  KeyVaultRoleDefinitionProperties properties) {
+    public KeyVaultRoleDefinition(String id, String name, KeyVaultRoleDefinitionType type, String roleName,
```

Comment:
Why publicly creatable?

Reply:
All these return types in Java have to be public so that deserialization can work properly. Even if users can create them they are not accepted as input parameters anywhere.

Reply:
Now that I think of it, given the way we create these types, the constructors can be package private.

Reply:
Actually, in this case it's because the model is in a different package than the client, which means we need to make the constructor and the class `public`.